frontend/src/components/pages/CreateReview/CreateReview.tsx

```diff
@@ -78,10 +98,90 @@ const CreateReview = (): React.ReactElement => {
     addBook(book);
   };
 
-  // useEffect hook adds dummy data to the books array
+  /**
+   * This function maps a list of AuthorResponse objects to a list
+   * of Author objects
+   *
+   * @param authorResponses - a list of AuthorResponse objects returned by the review API
+   * @returns a list of Author objects
+   */
+  const mapAuthorResponseToAuthor = (
+    authorResponses: AuthorResponse[],
+  ): Author[] => {
+    const result: Author[] = authorResponses.map((authorResponse) => ({
+      fullName: authorResponse.fullName,
+      displayName: authorResponse.displayName,
+      attribution: authorResponse.attribution,
+    }));
+
+    return result;
+  };
+
+  /**
+   * This function maps a list of Format objects to a list of
+   * BookFormat objects
+   *
+   * @param formatResponse - a list of Format objects returned by the review API
+   * @returns a list of BookFormat objects
+   */
+  const mapFormatToBookFormat = (
+    formatResponse: Format[] | null,
+  ): BookFormat[] => {
+    if (!formatResponse) {
+      return [];
+    }
+
+    const result: BookFormat[] = formatResponse.map((format) => ({
+      format: format.format,
+      price: format.price,
+      isbn: format.isbn,
+    }));
+
+    return result;
+  };
+
+  /**
+   * A callback function that maps a list of BookResponse objects
+   * to a list of Book objects and sets them in the component state
+   */
+  const setBooksFromBookResponse = useCallback(
+    (reviewResponse: ReviewResponse) => {
+      const result: Book[] = reviewResponse.books.map((response) => ({
+        title: response.title,
+        coverImage: response.coverImage,
+        titlePrefix: response.titlePrefix,
+        seriesOrder: response.seriesOrder,
+        illustrator: response.illustrator,
+        translator: response.translator,
+        formats: mapFormatToBookFormat(response.formats),
+        minAge: response.minAge,
+        maxAge: response.maxAge,
+        authors: mapAuthorResponseToAuthor(response.authors),
+        publishers: response.publishers,
+        seriesName: response.seriesName,
+      }));
+      setBooks(result);
+    },
+    [setBooks],
+  );
+
   useEffect(() => {
-    setBooks(data);
-  }, []);
+    if (id) {
+      reviewAPIClient
+        .getReviewById(id)
+        .then((reviewResponse: ReviewResponse) => {
+          setReview(reviewResponse.body);
```

Comment:
(minor) do you think using `await` would be cleaner here?

Reply:
they're basically the same - i saw other places do it this way so that's why I went this